src/features/messages/useMessageHistory.ts

```
@@ -0,0 +1,189 @@
+  hasMore: boolean;
+  loadMore: () => void;
+  refresh: () => void;
+}
+
+export function useMessageHistory(
+  walletAddresses: string[],
+  warpRouteAddresses: string[],
+  multiProvider: MultiProtocolProvider,
+): UseMessageHistoryResult {
+  const multiProviderRef = useRef(multiProvider);
+  multiProviderRef.current = multiProvider;
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/src/features/messages/useMessageHistory.ts:35:3
  33 | ): UseMessageHistoryResult {
  34 |   const multiProviderRef = useRef(multiProvider);
> 35 |   multiProviderRef.current = multiProvider;
     |   ^^^^^^^^^^^^^^^^^^^^^^^^ Cannot update ref during render
  36 |
  37 |   const walletKey = useMemo(() => JSON.stringify([...walletAddresses].sort()), [walletAddresses]);
  38 |   const warpRouteKey = useMemo(

(react-hooks/refs)